packages/react-components/react-positioning/src/createPositionManager.ts

```diff
@@ -69,7 +69,19 @@ export function createPositionManager(options: PositionManagerOptions): Position
   }
 
   // When the dimensions of the target or the container change - trigger a position update
-  const resizeObserver = disableUpdateOnResize ? null : createResizeObserver(targetWindow, () => updatePosition());
+  const resizeObserver = disableUpdateOnResize
+    ? null
+    : createResizeObserver(targetWindow, entries => {
+        // If content rect dimensions to go 0 -> very likely that `display: none` is being used to hide the element
+        // In this case don't update and let users update imperatively
+        const shouldUpdateOnResize = entries.every(entry => {
```

Comment:
TODO add test

Reply:
And mention it in storybook docs?

Reply:
I'm not sure, imperative positioning will still remain _the only way_ users can trigger than themselves in situations where the positioning is incorrect. We're simply adding some built in solution there, not sure what to document really.

As we've seen in Teams, auto repositioning when this case happens is actually incorrect since it always resulted in an incorrect state. I can't think of any situations either where it's desirable